student/src/main/java/tds/student/services/remote/RemoteOpportunityService.java

```diff
@@ -252,4 +316,41 @@ public void waitForSegment(final OpportunityInstance oppInstance, final int segm
   public void exitSegment(final OpportunityInstance oppInstance, final int segmentPosition) throws ReturnStatusException {
     legacyOpportunityService.exitSegment(oppInstance, segmentPosition);
   }
+
+  private static TestConfig mapExamConfigurationToTestConfig(ExamConfiguration examConfiguration) {
+    Exam exam = examConfiguration.getExam();
+    TestConfig testConfig = new TestConfig();
+    testConfig.setStatus(OpportunityStatusExtensions.parseExamStatus(examConfiguration.getStatus()));
+    testConfig.setRestart(exam.getRestartsAndResumptions());
+    testConfig.setTestLength(examConfiguration.getTestLength());
+    testConfig.setStartPosition(examConfiguration.getStartPosition());
+    testConfig.setContentLoadTimeout(examConfiguration.getContentLoadTimeoutMinutes());
+    testConfig.setInterfaceTimeout(examConfiguration.getInterfaceTimeoutMinutes());
+    testConfig.setRequestInterfaceTimeout(examConfiguration.getRequestInterfaceTimeoutMinutes());
+    testConfig.setOppRestartMins(examConfiguration.getExamRestartWindowMinutes());
+    testConfig.setPrefetch(examConfiguration.getPrefetch());
+    testConfig.setScoreByTDS(false);
+    testConfig.setValidateCompleteness(examConfiguration.isValidateCompleteness());
+    //TODO: set MSB flag
+    return testConfig;
+  }
+
+  private OpportunitySegment.OpportunitySegments mapExamSegmentsToOpportunitySegments(List<ExamSegment> examSegments) {
+    OpportunitySegment.OpportunitySegments opportunitySegments = new OpportunitySegment().new OpportunitySegments();
```

Comment:
I DO NOT like having to instantiate this silly inner-list class in this manner, but I could not find an alternative to doing it like this. Any suggestions are welcome.